src/text_engine/boundaries/Door.java

```diff
@@ -83,20 +83,10 @@ public Room getOtherRoom(Room from) throws IllegalArgumentException, IllegalStat
      */
     @Override
     public String toString() {
-        StringBuilder result = new StringBuilder();
-        if (room1 != null && room2 != null) {
-            result.append("Door between ")
-                  .append(room1.getName())
-                  .append(" and ")
-                  .append(room2.getName());
-        }
-        else if (room1 != null ^ room2 != null) {
-            result.append("Dead-end door connected to ");
-            result.append(room1 != null ? room1 : room2);
-        }
-        else {
-            result.append("Roomless door");
-        }
```

Comment:
Realised that one of our invariants in `Door` is that neither room can be `null`, so got rid of the logic around one/both of them being `null`.